dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/jobService/data/PutJobData.kt

```diff
@@ -45,40 +55,52 @@ data class PutJobData(private val items: List<Pair<String, Path>>,
 
     override fun runningTitle(): String {
         val transferringPut = jobTaskElement.resourceBundle.getString("transferringPut")
-        val jobId = job?.jobId
+        val jobId = job.jobId
         val startedAt = jobTaskElement.resourceBundle.getString("startedAt")
         val started = jobTaskElement.dateTimeUtils.format(getStartTime())
         return "$transferringPut $jobId $startedAt $started"
     }
 
-    override var jobId: UUID? = null
-    public override var cancelled: Supplier<Boolean>? = null
+
+    override val jobId: UUID by lazy { job.jobId }
     override fun client(): Ds3Client = jobTaskElement.client
 
-    override public var lastFile = ""
     override fun internationalize(labelName: String): String = jobTaskElement.resourceBundle.getString(labelName)
     override fun modifyJob(job: Ds3ClientHelpers.Job) {
         job.withMaxParallelRequests(jobTaskElement.settingsStore.processSettings.maximumNumberOfParallelThreads)
     }
 
-    override var job: Ds3ClientHelpers.Job? = null
-        get() {
-            if (field == null) {
-                field = if (writeJobOptions() == null) {
-                    Ds3ClientHelpers.wrap(jobTaskElement.client).startWriteJob(bucket, buildDs3Objects())
-                } else {
-                    Ds3ClientHelpers.wrap(jobTaskElement.client).startWriteJob(bucket, buildDs3Objects(), writeJobOptions())
-                }
-
-            }
-            jobId = field?.jobId
-            return field!!
-        }
-        set(value) {
-            if (value != null) {
-                field = value
+    override val job: Ds3ClientHelpers.Job by lazy {
+                val ds3Objects = items.map { dataToDs3Objects(it) }.flatMap { it.asIterable() }
+                ds3Objects.map { pair: Pair<Ds3Object, Path> -> Pair<String, Path>(pair.first.name, pair.second) }
+                    .forEach { prefixMap.put(it.first,it.second)}
+                val priority =
+                    if (jobTaskElement.savedJobPrioritiesStore.jobSettings.putJobPriority.equals("Data Policy Default (no change)")) {
```

Comment:
is string comparison our only choice here? it seems brittle

Reply:
We're ensured we will either get the string ""Data Policy Default (no change)" or a valid member of the enum.